### PR TITLE
Revert "Revert "Added H2O support to the SM (#49624)""

### PR DESCRIPTION
**Pull request by LemonInTheDark**
This reverts commit 69a9eb08ff6464381aa5e3728f046b60b8a2b450.

## About The Pull Request

Fixes the sm producing fusion.  
`dynamic_heat_modifier` and `power_transmission_bonus` never got a clean reset, and because they are global vars, kept gaining value. I've fixed this and tweaked 1 comment about max heat output.
This was a result of misunderstanding from both ghil and ninja, and is now thankfully fixed.

## Why It's Good For The Game

This shit is neat, should actually be in the game rather then in my fixing pile.

## Changelog
:cl:
add: The h2o interactions have been polished and readded. Fucking global variables.
/:cl:

### CODE REVIEW

**Comment by carshalash** [comment]
Did you test it.

**Comment by LemonInTheDark** [comment]
of course
I'll do it again just for you

**Comment by LemonInTheDark** [comment]
image
There we go.
Toasty, but not fusing. We good.

**Comment by LemonInTheDark** [comment]
dumbass.mp4

**Comment by Acer202** [comment]
This concept is still really silly. Why does a high water vapor content destabilize the engine, but a fire extinguisher soaking it down fix it? Makes no sense!

**Comment by LemonInTheDark** [comment]
:)

**Comment by nicbn** [comment]
Nice now change bee description

**Comment by LemonInTheDark** [comment]
> 
> 
> Nice now change bee description

???????????????????????

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -430,21 +437,58 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+		dynamic_heat_modifier += tritiumcomp * TRITIUM_HEAT_PENALTY
+		dynamic_heat_modifier += (pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus
+		dynamic_heat_modifier += n2comp * NITROGEN_HEAT_PENALTY
+		dynamic_heat_modifier += bzcomp * BZ_HEAT_PENALTY
+		dynamic_heat_modifier *= h2omalus
+		dynamic_heat_modifier += (h2ocomp * H2O_HEAT_PENALTY) * h2ofixed
+		dynamic_heat_modifier = max(dynamic_heat_modifier, 0.5)
+		//Value between 1 and 10
+		dynamic_heat_resistance = max((n2ocomp * N2O_HEAT_RESISTANCE) + ((h2ocomp * H2O_HEAT_RESISTANCE) * h2obonus) + ((pluoxiumcomp * PLUOXIUM_HEAT_RESISTANCE) * pluoxiumbonus), 1)
 		//Value between 30 and -5, used to determine radiation output as it concerns things like collecters
```
Comment (Dennok) [inline code/modules/power/supermatter/supermatter.dm]: Small grammar please.
```suggestion
		//Value between 30 and -5, used to determine radiation output as it concerns things like collectors
```

**Comment by Ghilker** [comment]
> This concept is still really silly. Why does a high water vapor content destabilize the engine, but a fire extinguisher soaking it down fix it? Makes no sense!

because the fire extinguisher emit a very low amount of water vapor, instead you need an high amount of h2o in the chamber for it to interact in that way

**Comment by LemonInTheDark** [comment]
> 
> 
> > This concept is still really silly. Why does a high water vapor content destabilize the engine, but a fire extinguisher soaking it down fix it? Makes no sense!
> 
> because the fire extinguisher emit a very low amount of water vapor, instead you need an high amount of h2o in the chamber for it to interact in that way

Something something different types of water something something time-green

The water you get out of extinguishers is a reagent iirc. H20 vapor is a gas. Reagents never become gas.

Even from an upfront view extinguishers don't work by cooling the engine, they just cool the air around it.

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: Thank ya :)

**Comment by kriskog** [comment]
Here we go aga-
Nope, merge conflict.

**Comment by LemonInTheDark** [comment]
God fucking damnit ghil stop being so productive.

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -141,10 +144,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	var/tritiumcomp = 0
 	var/bzcomp = 0
 	var/pluoxiumcomp = 0
+	var/h2ocomp = 0
```
Comment (spookydonut) [inline code/modules/power/supermatter/supermatter.dm]: code doc these

**Review by spookydonut — CHANGES_REQUESTED** [review]

File: code/modules/power/supermatter/supermatter.dm (hunk 2 above)
Comment (Ghilker) [reply code/modules/power/supermatter/supermatter.dm]: there is the code doc on the n2comp `// raw composition of each gas in the chamber, ranges from 0 to 1`

Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: fineeeeeeeeeeeeeeeeeeeeeeee

**Comment by jtroopa** [comment]
Honestly, water vapor as a reactant in the SM makes it one step closer to an actual nuclear reactor, and I approve every shaky step towards that singularity.
And adds an extra step in the SM fire reaction from plasma fire, to fusion, to tritium burning, to water vapor. I'd love for water vapor buildup to lead to something more catastrophic in that chain reaction. Maybe it'd give those dipshit engineers incentive not to let the SM burst into flames.

File: code/modules/power/supermatter/supermatter.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: I've done the value you requested and the ones directly related to it. I will cover the rest in a separate pr.

Comment (optimumtact) [reply code/modules/power/supermatter/supermatter.dm:148]: it's non negotiable to document vars you are adding, it must be done.

Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm:148]: Understood, I was refering to the sm as a whole. I will doc the other vars that were added here.

**Comment by LemonInTheDark** [comment]
I've turned ghil's docs into code docs, I apologize if I've missed anything or come off as confrontational.

**Comment by optimumtact** [comment]
no problem friend 👍

**Pull request merged**